### PR TITLE
Feature/sc 121028/get device os files to flash

### PR DESCRIPTION
**Pull request by hugomontero**
## Description
This PR will add device os binaries depending on the received flash params 

## Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed [CLA](https://docs.google.com/a/particle.io/forms/d/1_2P-vRKGUFg5bmpcKLHO_qNZWGi5HKYnfrrkd-sbZoA/viewform)
- [x] Problem and solution clearly stated
- [x] Tests have been provided
- [x] Docs have been updated
- [x] CI is passing

### CODE REVIEW

File: src/cmd/flash.js [hunk 1]
```diff
@@ -301,4 +311,43 @@ module.exports = class FlashCommand extends CLICommandBase {
 		}
 		return Array.from(binaries);
 	}
+
+	async _getDeviceOsBinaries({ skipDeviceOSFlash, target, files, firmwareVersion, platformId, applicationOnly }) {
+		const { particleApi } = this._particleApi();
+		const [binary] = files;
```
Comment (monkbroc) [inline src/cmd/flash.js:317]: You need to find the application module, not just pick the first file.

File: src/cmd/flash.js [hunk 2]
```diff
@@ -301,4 +311,43 @@ module.exports = class FlashCommand extends CLICommandBase {
 		}
 		return Array.from(binaries);
 	}
+
+	async _getDeviceOsBinaries({ skipDeviceOSFlash, target, files, firmwareVersion, platformId, applicationOnly }) {
```
Comment (monkbroc) [inline src/cmd/flash.js:315]: firmwareversion isn't passed by the caller. What is it for?

File: src/lib/device-os-version-util.js [hunk 3]
```diff
@@ -122,13 +122,17 @@ async function isModuleDownloaded(module, version, platformName) {
...
  */
-async function downloadDeviceOsVersionBinaries({ api, platformId, version='latest', ui }){
+async function downloadDeviceOsVersionBinaries({ api, platformId, version='latest', ui, omitUserPart=false }){
 	try {
 		const downloadedBinaries = [];
 		// get platform by id from device-constants
 		const platform = Object.values(deviceConstants).filter(p => p.public).find(p => p.id === platformId);
 		// get the device os versions
 		const deviceOsVersion = await api.getDeviceOsVersions(platformId, version);
+		if (omitUserPart) {
```
Comment (monkbroc) [inline src/lib/device-os-version-util.js:132]: I can't imagine why we'd want to download the user part ever. Can you always skip it?

**Review by monkbroc — CHANGES_REQUESTED** [review]
The biggest change is to make sure to grab the user application from the list of files.

**Comment by monkbroc** [comment]
Also if you rebase, Windows CI should pass

File: src/cmd/flash.js (hunk 1 above)
Comment (hugomontero) [reply src/cmd/flash.js]: Did it here! https://github.com/particle-iot/particle-cli/pull/657/commits/45be6c14118f41cf918575c8b02679d83783bf2d.

File: src/cmd/flash.js (hunk 2 above)
Comment (hugomontero) [reply src/cmd/flash.js:317]: I wasn't intended hehe. I was testing the function and I forgot to pass again the firmwareVersion. Fixed here! https://github.com/particle-iot/particle-cli/pull/657/commits/624460175e0fd070982c10a681cf8fc7b61a41d5

File: src/lib/device-os-version-util.js (hunk 3 above)
Comment (hugomontero) [reply src/lib/device-os-version-util.js]: I was thinking like the setup,  but specially here there is not a reason to not omit it always Fixed here! https://github.com/particle-iot/particle-cli/pull/657/commits/268d6e892a5a68e954f62c6e0d7e932c8299c4e5

**Review by monkbroc — APPROVED** [review]
:+1:

**Pull request merged**